Detectors/TPC/simulation/include/TPCSimulation/DigitPad.h

```diff
@@ -53,41 +60,35 @@ class DigitPad{
     /// \param pad pad ID
     void fillOutputContainer(TClonesArray *output, int cru, int timeBin, int row, int pad, float commonMode = 0);
 
-    /// The MC labels are sorted by occurrence such that the event/track combination with the largest number of occurrences is first
-    /// This is then dumped into a std::vector and attached to the digits
-    /// \todo Find out how many different event/track combinations are relevant
-    /// \param std::vector containing the sorted MCLabels
-    void processMClabels(std::vector<long> &sortedMCLabels);
-    
   private:
     float                  mChargePad;   ///< Total accumulated charge on that pad for a given time bin
     unsigned char          mPad;         ///< Pad of the ADC value
-    std::map<long, int>    mMCID;        ///< Map containing the MC labels (key) and the according number of occurrence (value)
+    FairMultiLinkedData    mMCLinks;     ///< MC links
```

Comment:
DigitPad is our own sorting container and does not inherit from from FairMultiLinkedData_Interface.
`class DigitPad{`
When we format the output data (DigitMC) we use exactly this inheritance:
`
void DigitPad::fillOutputContainer(TClonesArray *output, int cru, int timeBin, int row, int pad, float commonMode)
{
  /// The charge accumulated on that pad is converted into ADC counts, saturation of the SAMPA is applied and a Digit is created in written out
  const float totalADC = mChargePad - commonMode; // common mode is subtracted here in order to properly apply noise, pedestals and saturation of the SAMPA

  const float mADC = SAMPAProcessing::makeSignal(totalADC, PadSecPos(CRU(cru).sector(), PadPos(row, pad)));
  if(mADC > 0) {
    TClonesArray &clref = *output;
    const size_t digiPos = clref.GetEntriesFast();
    DigitMC *digit = new(clref[digiPos]) DigitMC(cru, mADC, row, pad, timeBin, commonMode);
    digit->SetLinks(getMCLinks());
  }
}
`